src/base/error_manager.h

```diff
@@ -85,6 +85,86 @@
     } \
   while (0)
 
+#define ERROR_SET_WARNING(error, code) \
+  do \
+    { \
+      error = code; \
+      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, code, 0); \
+    } \
+  while (0)
```

Comment:
How about using a variable argument macro?

```
#define ERROR_SET_WARNING(error, code, arg_cnt, ...) \
  do \
    { \
      error = code; \
      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, code, arg_cnt, ##__VA_ARGS__); \
    } \
  while (0)
```
  
or

```
#define ERROR_SET_WARNING(error, code, arg_cnt, args...) \
  do \
    { \
      error = code; \
      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, code, arg_cnt, ##args); \
    } \
  while (0)  

```

Reply:
Not bad, but isn't it the same as calling the er_set function directly?

Reply:
It's a way to reduce multiple macro functions into one, but I'll keep it that way here.